### PR TITLE
feat(scheduling): [MC-859] Provide a reason for unscheduling an item on New Tab DE

### PR DESCRIPTION
**Pull request by nina-py**
## Goal

Provide a reason for unscheduling an item on New Tab DE.

- Enabled unscheduling/removal reasons for NEW_TAB_DE_DE

- Updated tests.

## Video walkthrough

https://github.com/Pocket/curation-admin-tools/assets/22447785/d4f57dc9-d558-4616-9f24-c5fb9a12924f

## Reference

https://mozilla-hub.atlassian.net/browse/MC-859

### CODE REVIEW

File: src/curated-corpus/components/RemoveItemFromScheduledSurfaceModal/RemoveItemFromScheduledSurfaceModal.test.tsx [hunk 1]
```diff
@@ -1,32 +1,36 @@
...
+} from '../../../api/generatedTypes';
 import { RemoveItemFromScheduledSurfaceModal } from './RemoveItemFromScheduledSurfaceModal';
 import { getTestApprovedItem } from '../../helpers/approvedItem';
-import { ScheduledSurfaces } from '../../helpers/definitions';
 import userEvent from '@testing-library/user-event';
 
 describe('The RemoveItemFromScheduledSurfaceModal component', () => {
   const approvedItem = getTestApprovedItem();
-  const germanItem: ScheduledCorpusItem = {
+  const pocketHitsItem: ScheduledCorpusItem = {
```
Comment (nina-py) [inline src/curated-corpus/components/RemoveItemFromScheduledSurfaceModal/RemoveItemFromScheduledSurfaceModal.test.tsx:13]: For the form version without reasons for unscheduling, I switched the surface to one of the Pocket Hits ones.

File: src/curated-corpus/components/RemoveItemFromScheduledSurfaceModal/RemoveItemFromScheduledSurfaceModal.test.tsx [hunk 2]
```diff
@@ -1,32 +1,36 @@
...
     approvedItem,
     createdAt: 1635014926,
     createdBy: 'jdoe',
     externalId: '123-test-id',
     scheduledDate: '2022-05-24',
-    scheduledSurfaceGuid: ScheduledSurfaces[1].guid,
+    scheduledSurfaceGuid: 'POCKET_HITS_DE_DE',
```
Comment (nina-py) [inline src/curated-corpus/components/RemoveItemFromScheduledSurfaceModal/RemoveItemFromScheduledSurfaceModal.test.tsx:19]: Updated test data for readability

File: src/curated-corpus/components/RemoveItemFromScheduledSurfaceModal/RemoveItemFromScheduledSurfaceModal.test.tsx [hunk 3]
```diff
@@ -1,32 +1,36 @@
...
     updatedAt: 1635114926,
     updatedBy: 'jdoe',
+    source: ScheduledItemSource.Manual,
```
Comment (nina-py) [inline src/curated-corpus/components/RemoveItemFromScheduledSurfaceModal/RemoveItemFromScheduledSurfaceModal.test.tsx:22]: This was missing and erroring in the IDE but tests run fine without the recently introduced required property.

**Comment by efixler** [comment]
Changes look good.

I can understand moving the formerly German tests to Pocket Hits, but why no test for the modified German surface?

**Comment by efixler** [comment]
There's also a failing test and a bunch of warnings in the unit tests.

**Review by katerinachinnappan — APPROVED** [review]

**Comment by nina-py** [comment]
@efixler, testing either the US or the German surface is enough - the tests use the US New Tab GUID. I don't think we should write tests for every surface with additional reasons enabled - we generally don't do that for other functionality that is only available for a subset of surfaces/other parameters.

I'm very aware of the warnings in the unit tests! Some are from deprecated packages and some are from deprecated components/component parameters in MUI that we also need to update (and the Great App Update will take care of that). The one failing test is the flaky time-zone sensitive one that has been the bane of this repository for a  good part of two years.

**Pull request merged**